### ISSUE TITLE
dune pkg lock fails on newly created project

### ISSUE THREAD

**Issue by finalclass** [ocaml/dune#11057]
## Reproduction

```
$ dune init project test5
Entering directory '/home/sel/Documents/tmp/test5'
Success: initialized project component named test5
~/Documents/tmp (master *)$ cd test5/
~/Documents/tmp/test5 (master *)$ dune pkg lock
Solution for dune.lock:
- base-threads.base
- base-unix.base
- dune.3.16.0
- ocaml.5.1.1
- ocaml-base-compiler.5.1.1~rc1
- ocaml-config.3
$ dune build
Error: Multiple rules generated for
_build/_private/default/.pkg/ocaml-base-compiler/source:
- dune.lock/ocaml-base-compiler.pkg:24
- dune.lock/ocaml-base-compiler.pkg:13
-> required by - package ocaml-config
-> required by - package ocaml
```

## Specifications

```
$ dune --version
3.16.0
$ ocaml --version
The OCaml toplevel, version 5.2.1+dev0-2024-05-13
$ cat /etc/lsb-release
DISTRIB_ID="ManjaroLinux"
DISTRIB_RELEASE="24.1.1"
DISTRIB_CODENAME="Xahea"
DISTRIB_DESCRIPTION="Manjaro Linux"
$ uname -a
Linux rog 5.15.167-1-MANJARO #1 SMP PREEMPT Thu Sep 12 11:38:41 UTC 2024 x86_64 GNU/Linux
```

I tried also on ocaml 5.1.0 with the same result.

**Comment by maiste** [comment]
Hi, thanks for reporting :+1: 
Did you use the latest stable or the https://preview.dune.build version? If you used `3.16` many of the changes from package management are not available there. Two solutions exist:
- Using the preview binary at https://preview.dune.build where the flags are enabled.
- Using the version from the `main` branch, but in this case you need to activate the features with `DUNE_CONFIG__TOOLCHAINS=enabled dune ...`.

**Comment by finalclass** [comment]
Ah, I thought that if `dune pkg lock` was available in 3.16.0, then it should work. :)
I tried this in dune nightly and it worked as expected. I was using version 3.16.0.

This feature looks amazing! I can't wait to use it.
